src/v/cluster/partition.h

```diff
@@ -93,20 +93,38 @@ class partition {
     }
 
     ss::future<result<model::offset, std::error_code>>
-    sync_effective_start(model::timeout_clock::duration timeout) {
+    sync_kafka_start_offset_override(model::timeout_clock::duration timeout) {
         if (_log_eviction_stm) {
-            co_return co_await _log_eviction_stm->sync_effective_start(timeout);
+            auto offset_res = co_await _log_eviction_stm
+                                ->sync_kafka_start_offset_override(timeout);
+            if (offset_res.has_failure()) {
+                co_return offset_res.as_failure();
+            }
+            auto offset_translator_state = get_offset_translator_state();
+            if (_raft->start_offset() < offset_res.value()) {
+                auto start_kafka_offset
+                  = offset_translator_state->from_log_offset(
```

Comment:
Theres alot of to kafka offset translations in this class, also changes to replicated_partition too, can we move this logic to replicated_partition entirely?

Reply:
I found it clunky to move, exposing different kinds of offsets from different STMs. If you have suggestions on what interface we should expose at cluster/partition, I'm open to it, but I will keep this as is otherwise.

FWIW, we do have other pieces of kafka API sprinkled in this class when we need to straddle local and cloud storage, e.g. in timequeries.